util/stream.go

```diff
@@ -87,17 +87,18 @@ func (m *MessageStream) outbound() {
 		case <-m.Shutdown:
 			log.Infof("Closing OpenFlow message stream.")
 			m.conn.Close()
-			for i := 0; i < numParserGoroutines; i++ {
-				m.parserShutdown <- true
-			}
-
 			// clear Outbound chan
-			for {
-				select {
-				case <-m.Outbound:
-				case <-time.After(time.Minute * 2):
-					return
+			go func() {
+				for {
+					select {
+					case <-m.Outbound:
+					case <-time.After(time.Minute * 2):
+						return
+					}
 				}
+			}()
+			for i := 0; i < numParserGoroutines; i++ {
+				m.parserShutdown <- true
 			}
```

Comment:
Should return to break outbound go routine after send shutdown signal to parser go routines.  Other wise, this go coroutine may remained.